v2/pink-sb/src/lib/step/StepItem.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
+    import ActiveIndicator from './active-indicator.svg';
+    import Done from './done.svg';
+    import { Tag } from '$lib/index.js';
+    export let state: 'previous' | 'current' | 'next';
+    export let noLine: boolean = false;
+    export let shortLine: boolean = false;
+    export let hideBadge: boolean = false;
+</script>
+
+<div class="stepitem" class:noline={noLine} class:shortline={shortLine}>
+    {#if state === 'previous' || state === 'next'}
+        <div class="dot-inactive" />
+    {:else if state === 'current'}
+        <div class="dot-active">
+            <img src={ActiveIndicator} alt="Active" />
+        </div>
+    {/if}
+    {#if !hideBadge}
+        <div class="badge">
+            <Tag size="s">
+                {#if state === 'next'}
+                    Next
+                {:else if state === 'current'}
+                    Now
+                {:else}
+                    <img src={Done} alt="Done icon" /> <span>Done</span>
+                {/if}
+            </Tag>
+        </div>
+    {/if}
+    <div class:badge-margin={!hideBadge}>
+        <slot />
+    </div>
+</div>
+
+<style lang="scss">
+    .stepitem {
+        padding: 24px 29px;
+        position: relative;
+
+        &::before {
+            content: '';
+            position: absolute;
+            background-color: var(--color-light-neutral-10, #ededf0);
+            width: 1px;
+            height: 100%;
+            margin-left: -30px;
```

Comment:
Let's fututre proof this and use `margin-inline` and `margin-block` when possible

Reply:
Ayy nice, you learn every day! Done!